nc_test/tst_nofill.c

```diff
@@ -486,8 +486,8 @@ main(int argc, char **argv)
        if (nc_close(ncid2)) ERR;
        SUMMARIZE_ERR;
    }
-   FINAL_RESULTS;
 #ifdef USE_PNETCDF
    MPI_Finalize();
 #endif
+   FINAL_RESULTS;
```

Comment:
Yes, good catch. WIthout this change the program exits without calling MPI_Finalize() in parallel runs.